pennylane/interfaces/execution.py

```diff
@@ -618,11 +633,21 @@ def post_processing(results):
 
     _grad_on_execution = False
 
-    if config.use_device_gradient:
+    if config.use_device_jacobian_product and interface in jpc_interfaces:

```

Comment:
if i'm being honest.. my brain kinda panics from this line until line 778. the deeply nested if-else block does not make it clear what each scenario is, and I'm not sure what would. perhaps standardizing what goes in (idk) and out (`execute_fn` and `jpc`, right?) of each conditional block?

Reply:
I worked this into the existing nested if-else framework as I thought it might be easier to see the parallels between the two systems.

Instead, I could have one nested if-elif block for the new system and a separate block for the old system?

Reply:
I'm not sure what would be best tbh. the pylint comments at the top of the file make me lose a little hope for it, but maybe that would help? or extracting code into helper functions? I think I'd most want to see the nesting reduced, so whatever achieves that. dare I even say.. repeating parts of conditions to remove nesting (like `if a and b; do x; elif a and c; do y`)